### PR TITLE
[MM-34921] Add app icon suppport

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This is a draft for app icon support.

See https://community-daily.mattermost.com/core/pl/id3uus4u9brsbmjedc7sf6n5yw for the discussion.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34921
https://mattermost.atlassian.net/browse/MM-35376

### CODE REVIEW

File: server/config/config.go
Comment (hanzei) [reply server/config/config.go]: Good catch. `path.Join` should only be used for the path, not the schema. Fixed in https://github.com/mattermost/mattermost-plugin-apps/pull/196/commits/1900bbc685afd3abf40efa11b9a2a9350267e2da.

File: server/proxy/proxy.go
Comment (hanzei) [inline server/proxy/proxy.go:153]: The PR won't work because static assets are only work for installed apps. Before I change that, I wanted to get approval for the idea.

Comment (hanzei) [reply server/proxy/proxy.go]: Fixed with [`c0c6e18` (#196)](https://github.com/mattermost/mattermost-plugin-apps/pull/196/commits/c0c6e18a95528aab5180272e6b02582cd555dbac)

**Review by levb — APPROVED** [review]
Not sure what was the need/usecase for `StaticUpstream` as a separate interface, but I'm ok with it

**Review by larkox — APPROVED** [review]
LGTM

**Comment by hanzei** [comment]
@levb `StaticUpstream` is needed for `staticUpstreamForManifest` as I wanted a uniform method that returns an upstream based on a manifest.

**Comment by hanzei** [comment]
Heads up that I moved the Marketplace icon URL generation server side in [`b7880fe` (#196)](https://github.com/mattermost/mattermost-plugin-apps/pull/196/commits/b7880fedb67cee586e1da7a9e42c361eb67f6522).

**Comment by DHaussermann** [comment]
@jfrerich is there a Zendesk build or PR that can be used to validate this change?

**Comment by hanzei** [comment]
@DHaussermann Yes, you can use https://github.com/mattermost/mattermost-app-servicenow/pull/32 or https://github.com/mattermost/mattermost-app-zendesk/pull/61.

**Comment by hanzei** [comment]
@DHaussermann FYI: @jfrerich did already test https://github.com/mattermost/mattermost-app-zendesk/pull/61

**Comment by jfrerich** [comment]
@DHaussermann, do you want to retest this, or can I merge?

**Comment by DHaussermann** [comment]
@jfrerich given your peer-testing, please merge now.  l'll double check for any issues on release testing.

LGTM!

**Review by DHaussermann — APPROVED** [review]

**Pull request merged**
